pkg/pulumiyaml/codegen/gen_program.go

```diff
@@ -71,13 +71,26 @@ func GenerateProject(directory string, project workspace.Project, program *pcl.P
 	if err != nil {
 		return err
 	}
-	files["Pulumi.yaml"] = projectBytes
+	// Pulumi.yaml always needs to be written to the root directory
+	err = os.WriteFile(path.Join(directory, "Pulumi.yaml"), projectBytes, 0600)
```

Comment:
so files is already returned above from `GenerateProgram` so we'd have to go through all the keys in that map and reroot them to `path.Join(directory, project.Main)`. Seemed a lot simpler to write Pulumi.yaml by itself then do that re-root in the for loop below rather than mutating the map.